src/common/tests_dmg_helpers.c

```diff
@@ -94,6 +95,136 @@ cmd_string(const char *cmd_base, char *args[], int argcount)
 	return cmd_str;
 }
 
+static void
+log_stderr_pipe(int fd)
+{
+	char	buf[512];
+	char	*full_msg = NULL;
+	ssize_t	len = 0;
+
+	D_DEBUG(DB_TEST, "reading from stderr pipe\n");
+	while (1) {
+		ssize_t n;
+		ssize_t old_len = len;
+		char	*tmp;
+
+		n = read(fd, buf, sizeof(buf));
+		if (n == 0)
+			break;
+		if (n < 0) {
+			D_ERROR("read from stderr pipe failed: %s\n", strerror(errno));
+			break;
+		}
+
+		len = len + n;
+		D_REALLOC(tmp, full_msg, old_len, len);
+		if (tmp == NULL) {
+			D_ERROR("reading from stderr pipe: can't realloc tmp with size %ld\n",
+				len);
+			break;
+		}
+
+		full_msg = tmp;
+		strncpy(&full_msg[old_len], buf, n);
```

Comment:
Maybe a memcpy could be more suited than a strncpy if the buffer contains multiple '\0' (for example used as separator).

Reply:
In this case we don't expect there to be any null characters until the end of the string.